lib/auth/clt.go

```diff
@@ -381,11 +411,7 @@ func (c *Client) GetClusterCACert() (*LocalCAResponse, error) {
 }
 
 func (c *Client) Close() error {
-	c.Lock()
-	defer c.Unlock()
-	if c.transport != nil {
-		c.transport.CloseIdleConnections()
-	}
+	c.HTTPClient.Close()
 	return c.APIClient.Close()
```

Comment:
`c.APIClient` is no longer closed under a mutex.
is that safe?

maybe move `HTTPClient.mux` over to `Client`?

Reply:
IIUC, `APIClient.Close()` is already thread safe due to the use of an atomic flag. I remember in a previous PR I kept this mutex only for the HTTP Client. Now that I am removing the transport nil check and `CloseIdleConnections` is also thread safe, This client has no need for a mutex at all.